mail_tracking/models/mail_tracking_email.py

```diff
@@ -103,10 +103,13 @@ def email_is_bounced(self, email):
 
     @api.model
     def email_score_from_email(self, email):
-        if email:
-            return self.search([
-                ('recipient_address', '=', email.lower())]).email_score()
-        return 0.
+        if not email:
+            return 0.
+        data = self.read_group([('recipient_address', '=', email.lower())],
+                               ['recipient_address', 'state'], ['state'])
+        mapped_data = dict([(state['state'], state['state_count'])
+                            for state in data])
```

Comment:
Remove the surrounding `[...]` brackets.

With brackets:

1. Produce a lazy generator.
2. Exhaust the generator converting it into a list.
3. Convert the list in a dict.

Without brackets:

1. Produce a lazy generator.
2. Exhaust the generator converting it into a dict.

You can save tons of loops! :blush: 

---

Note that another idiomatic way (equivalent to doing it without brackets) to do so is: 

```py
mapped_data = {state['state']: state['state_count'] for state in data}
```

Reply:
Noted and fixed :wink:

Full coverage as well.